// dxy-wechat-share/src/dxy-wechat-share.js
var jQuery;
if (typeof require !== 'undefined') {
    jQuery = require('jquery');
    var wx = require('weixin-js-sdk');
}
else {
    jQuery = $;  // for IE8
}

//在微信中分享

(function ($) {
    function dxy_wechat_share(options) {

        var wxConfig = function (data) {
            var config;
            var siteConfig;
            var shareConfig;

            config = {
                debug: options.debug || false,
                appId: data.appid,
                timestamp: data.timestamp,
                nonceStr: data.noncestr,
                signature: data.sign,
                jsApiList: ['onMenuShareTimeline', 'onMenuShareAppMessage', 'onMenuShareQQ', 'onMenuShareWeibo']
            };

            // 站点的微信分享的 WX_CONFIG
            siteConfig = window.WX_CONFIG || {
                    title: options.title,
                    desc: options.desc,
                    imgUrl: options.imgUrl,
                    success: options.success,
                    cancel: options.cancel
                };

            shareConfig = {
                title: siteConfig.title || document.title,
                desc: siteConfig.desc || document.title,
                imgUrl: siteConfig.imgUrl,
                success: siteConfig.success,
                cancel: siteConfig.cancel,
                link: window.location.href
            };

            wx.config(config);

            wx.ready(function () {
                wx.onMenuShareTimeline(shareConfig);
                wx.onMenuShareAppMessage(shareConfig);
                wx.onMenuShareQQ(shareConfig);
                wx.onMenuShareWeibo(shareConfig);
            });
        };

        if (window.wx_share) {
            var simApi = window.wx_share;
            $.getJSON(simApi, wxConfig);
        }
        else {
            console.error('window.wx_share不存在');
        }
    }

    if (typeof module !== 'undefined' && typeof module.exports !== 'undefined') {
        module.exports = dxy_wechat_share;
    }
    else {
        window.dxy_wechat_share = dxy_wechat_share;
    }
})(jQuery);
